torch/csrc/jit/export.cpp

```diff
@@ -649,11 +833,23 @@ void ScriptModuleSerializer::convertModule(
     torch::ParameterDef* param_def = module_def->add_parameters();
     convertParameter(elem.value(), param_def, /*is_buffer=*/false);
   }
-  for (const auto& elem : module.get_attributes()) {
-    if (elem.value().type->isSubtypeOf(TensorType::get())) {
-      torch::ParameterDef* param_def = module_def->add_parameters();
-      convertParameter(elem.value(), param_def, /*is_buffer=*/true);
+
+  if (module.get_attributes().size() > 0) {
+    Pickler p(*this);
+    p.start();
+    for (const auto& item : module.get_attributes()) {
+      auto& attribute = item.value();
+
+      // Add attribute to ModuleDef
+      torch::AttributeDef* attribute_def = module_def->add_attributes();
+      attribute_def->set_name(attribute.name_);
+      attribute_def->set_type(attribute.type->python_str());
+
+      // Add attribute to pickle blob
+      p.addIValue(*attribute.slot());
     }
+    p.finish();
+    writer_.writeRecord("attributes", p.stack().data(), p.stack().size());
```

Comment:
This will clobber another submodules records. There should be one global attributes file written at the end of the model.